Source/SwiftLintFramework/Rules/Idiomatic/TypeNameRule.swift

```diff
@@ -92,3 +94,13 @@ public struct TypeNameRule: ASTRule, ConfigurationProviderRule {
         return []
     }
 }
+
+fileprivate extension String {
+    func nameStrippingTrailingSwiftUIPreviewProvider(_ dictionary: [String: SourceKitRepresentable]) -> String {
+        guard dictionary.inheritedTypes.contains("PreviewProvider"),
+            let lastPreviewsIndex = lastIndex(of: "_Previews")
```

Comment:
I'd make sure that `_Previews` is also the suffix just to be extra careful

Reply:
Okay.  Done.  I'm just waiting for a response to this [comment](https://github.com/realm/SwiftLint/pull/2815/files/1f73dcd0edbfa2a00a69875a505c68583c450078#r317406376) to push the changes